test/test_nestedtensor.py

```diff
@@ -7356,6 +7356,38 @@ def g(nt):
         output.backward(torch.ones_like(output))
         self.assertEqual(output._metadata_cache, cache)
 
+    # See https://github.com/pytorch/pytorch/issues/128649
+    @xfailIfTorchDynamo
+    @dtypes(torch.float32)
+    def test_composite_op_in_inference_mode(self, device, dtype):
+        # expect view
+        nt = random_nt_from_dims(
+            [4, None, 48],
+            device=device,
+            dtype=dtype,
+            layout=torch.jagged,
+            requires_grad=True,
+        )
+
+        with torch.inference_mode():
```

Comment:
just checking, this PR doesn't fix the issue when there is a custom TorchDispatchMode right?

Reply:
that's correct

Reply:
Thanks. Maybe we can add a unit case test that cover the custom TorchDispatchMode in test_nestedtensor?

Reply:
Sure, I'll add an NJT specific test as well :)